### DISCUSSION TITLE
QOL Improvement: No-entry sign default to "no entry".

### DISCUSSION THREAD

**Discussion by grahambilley** [OpenRCT2/OpenRCT2#25903] [Feature requests]
**OS:** [Windows 10]
**Version:** [0.2.4]
**Commit/Build:** [d645338]

This is an enhancement request. I'm putting this here per request of Aaron van Geffen.

When placing no-entry signs, have them default to "no-entry" mode (toggle on/off in settings). I like to make long path loops where flow only moves in one direction, and that works best with a no-entry sign on every tile. This would be extremely helpful.

Thank you,

**Comment by AaronVanGeffen** [comment]
To be honest, I‘m not convinced we should change the default behaviour. I don’t doubt it would be advantageous to your workflow, as you said. However, I’m afraid it would break the workflow of other users, who might use banners as, well, banners.

I think we should only change this if supported by numbers. Perhaps this can be part of a series of questions in a new survey to the community.

**Comment by grahambilley** [comment]
That makes sense to me. If the group is split and there are a lot of people who would like them to default to "no-entry", perhaps we could put a checkbox in the options menu?

**Comment by Umdlye** [comment]
Maybe this could work as a plug-in.

**Comment by ocalhoun6** [comment]
Maybe one option could be to duplicate every banner-type object in the scenery selection, having 'banner' and 'no entry banner' versions of each? (With each type, you could select the banner after placing it and change it. Selecting the 'no entry banner' type object in the scenery window would just make it default to a no-entry banner.) Note, I'm not talking about actually making additional objects -- I mean having them kind of as virtual objects that aren't really part of the save file, just something the scenery window extrapolates from the actual banner objects loaded in the particular scenario.

I think this would be a good way to implement it without cluttering anything up -- it would just slightly expand the objects in the scenery selection window. No truly new interface options. Also, it might help shallow out the learning curve for newer players who don't know about no-entry banners or how to use them. If they see a 'no entry banner' option in the scenery selection, they'll immediately know it's possible and have an easy way to do it.

**Comment by KatieZeldaKat** [comment]
Instead of duplicating the banner objects, I could see you being able to set the toggle for no-entry before placing it down, much like you are able to set the color of scenery before placing it. The symbol would show up on the right side of the scenery window if a banner is selected.

**Comment by karst** [comment]
> Instead of duplicating the banner objects, I could see you being able to set the toggle for no-entry before placing it down, much like you are able to set the color of scenery before placing it. The symbol would show up on the right side of the scenery window if a banner is selected.

I definitely like that approach.
Perhaps the only concern would be if we were to add more functions to the banner. For example possibly with the introduction of pools to have people in swimgear/clothes seperated, but that's a concern for the distant future.

**Comment by ocalhoun6** [comment]
>possibly with the introduction of pools to have people in swimgear/clothes seperated

Seems like that wouldn't necessarily require banners at all.

A) Don't have peeps change clothes at all. They just swim in their regular clothes.

B) Automatically change peeps to the 'swimwear' look as soon as they touch water. (And automatically change back the moment they leave the water.)

C) Peeps will automatically refuse to enter water if they're not wearing swimwear.

D) A swimming pool is dealt with like a special kind of ride, and the swimming peeps are treated like ride vehicles. (Honestly, a reskinned bumper cars/flying saucers ride could already *almost* do this. Make the floor of it look like water, and change the vehicles to be swimming peeps. Only tricky bit is the part where the peeps walk to and from the vehicles.)
